use of undeclared identifier 'sorespoQ_tmfQ___test_main'

### Acton Version

0.26.0.20251102.2.3.3

### Steps to Reproduce and Observed Behavior

sorespo builds in CI fail even though they build just fine locally on Mac and Linux (`acton build`), on the same acton version

GH Action log:
```
  Final compilation step
ERROR: internal compiler error: compilation of generated Zig code failed, returned error code1
NOTE: this is likely a bug in actonc, please report this at:
NOTE: https://github.com/actonlang/acton/issues/new?template=ice.yaml
NOTE: acton 0.26.0.20251102.2.3.3
zig stdout:

zig stderr:
Acton Project Builder - building /home/runner/work/sorespo/sorespo
Acton Base Builder
Building in /usr/lib/acton/base
-- filename : fs.c
   full_path: /usr/lib/acton/base/out/types/fs.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /fs.c
-- filename : re.c
   full_path: /usr/lib/acton/base/out/types/re.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /re.c
-- filename : process.c
   full_path: /usr/lib/acton/base/out/types/process.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /process.c
-- filename : __builtin__.c
   full_path: /usr/lib/acton/base/out/types/__builtin__.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /__builtin__.c
-- filename : http.c
   full_path: /usr/lib/acton/base/out/types/http.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /http.c
-- filename : net.c
   full_path: /usr/lib/acton/base/out/types/net.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /net.c
-- filename : xml.c
   full_path: /usr/lib/acton/base/out/types/xml.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /xml.c
-- filename : file.c
   full_path: /usr/lib/acton/base/out/types/file.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /file.c
-- filename : md5.c
   full_path: /usr/lib/acton/base/out/types/crypto/hash/md5.c
   dir      : /usr/lib/acton/base/out/types/crypto/hash
   file_path: /crypto/hash/md5.c
-- filename : json.c
   full_path: /usr/lib/acton/base/out/types/json.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /json.c
-- filename : wyhash.c
   full_path: /usr/lib/acton/base/out/types/hash/wyhash.c
   dir      : /usr/lib/acton/base/out/types/hash
   file_path: /hash/wyhash.c
-- filename : testing.c
   full_path: /usr/lib/acton/base/out/types/testing.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /testing.c
-- filename : qcheck.c
   full_path: /usr/lib/acton/base/out/types/qcheck.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /qcheck.c
-- filename : logging.c
   full_path: /usr/lib/acton/base/out/types/logging.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /logging.c
-- filename : uri.c
   full_path: /usr/lib/acton/base/out/types/uri.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /uri.c
-- filename : base64.c
   full_path: /usr/lib/acton/base/out/types/base64.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /base64.c
-- filename : argparse.c
   full_path: /usr/lib/acton/base/out/types/argparse.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /argparse.c
-- filename : diff.c
   full_path: /usr/lib/acton/base/out/types/diff.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /diff.c
-- filename : buildy.c
   full_path: /usr/lib/acton/base/out/types/buildy.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /buildy.c
-- filename : term.c
   full_path: /usr/lib/acton/base/out/types/term.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /term.c
-- filename : random.c
   full_path: /usr/lib/acton/base/out/types/random.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /random.c
-- filename : time.c
   full_path: /usr/lib/acton/base/out/types/time.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /time.c
-- filename : math.c
   full_path: /usr/lib/acton/base/out/types/math.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /math.c
-- filename : snappy.c
   full_path: /usr/lib/acton/base/out/types/snappy.c
   dir      : /usr/lib/acton/base/out/types
   file_path: /snappy.c
-- filename : rts.c
   full_path: /usr/lib/acton/base/out/types/acton/rts.c
   dir      : /usr/lib/acton/base/out/types/acton
   file_path: /acton/rts.c
Debug build
Threads enabled
Acton Project Builder - building /home/runner/.cache/acton/deps/orchestron-1220444d133e1e699488ae0a89d1366a37b2ecee34b9ee3ccc80875562b847484bfb
Acton Project Builder - building /home/runner/.cache/acton/deps/netconf-1220e8563054d5f3865584ba45e5777c652568084e6fac95c58933896228d70ff2af
Debug build
Threads enabled
Threads enabled
Acton Project Builder - building /home/runner/.cache/acton/deps/yang-1220d543befd132ee442f4dd39b50ffc8b6e209884dd69f7a10f80a31a630216059b
Debug build
Threads enabled
Debug build
Threads enabled
Acton Project Builder - building /home/runner/.cache/acton/deps/yang-122057d659b71c522133f8cf353aee1068526ae7ac8c5b32d74a625e9a7b6c753ecd
Debug build
Threads enabled
Acton Project Builder - building /home/runner/.cache/acton/deps/netconf-122001998709cee8d69937df65ebb51ace2dfc49897409d620f647f75d4b9775735e
Debug build
Threads enabled
Acton Project Builder - building /home/runner/.cache/acton/deps/actmf-1220270ffb06b469abacceae7749d1bd034f5ffd6960a5868b559607e002b13f36a0
Debug build
Threads enabled
Acton Project Builder - building /home/runner/.cache/acton/deps/rtrsecrets-1220fc5ebc12b8ae7fee4dae6ba674e49e6ef9421ca0484b36c9a138eb5c428fb0eb
Debug build
Threads enabled
Debug build
Threads enabled
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo.root.c -> sorespo
Building executable from: /home/runner/work/sorespo/sorespo/out/types/test_sorespo.test_root.c -> .test_test_sorespo
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo/test_rfs.test_root.c -> .test_sorespo.test_rfs
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c -> .test_sorespo.tmf
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo/crpd_stats.test_root.c -> .test_sorespo.crpd_stats
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo/cisco_vigenere.test_root.c -> .test_sorespo.cisco_vigenere
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo/test_inter.test_root.c -> .test_sorespo.test_inter
Building executable from: /home/runner/work/sorespo/sorespo/out/types/sorespo.test_root.c -> .test_sorespo
install
+- install .test_sorespo.tmf
   +- zig build-exe .test_sorespo.tmf Debug x86_64-linux-gnu.2.27 7 errors
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:30: error: use of undeclared identifier 'sorespoQ_tmfQ___test_main'
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:20: error: use of undeclared identifier '$t'
/home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490/rts/rts.h:224:32: note: expanded from macro '$NEWACTOR'
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:20: error: use of undeclared identifier 'sorespoQ_tmfQ___test_mainG_methods'
/home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490/rts/rts.h:224:46: note: expanded from macro '$NEWACTOR'
<scratch space>:256:1: note: expanded from here
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:20: error: use of undeclared identifier '$t'
/home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490/rts/rts.h:225:65: note: expanded from macro '$NEWACTOR'
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:20: error: use of undeclared identifier '$t'
/home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490/rts/rts.h:226:32: note: expanded from macro '$NEWACTOR'
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:20: error: use of undeclared identifier '$t'
/home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490/rts/rts.h:229:32: note: expanded from macro '$NEWACTOR'
/home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c:7:20: error: operand of type 'void' where arithmetic or pointer type is required
/home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490/rts/rts.h:223:29: note: expanded from macro '$NEWACTOR'
error: the following command failed with 7 compilation errors:
/usr/lib/acton/zig/zig build-exe -cflags -ffile-prefix-map=/home/runner/work/sorespo/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /home/runner/work/sorespo/sorespo/out/types/sorespo/tmf.test_root.c /home/runner/.cache/acton/zig-local-cache/o/ff930f8e23ef50fd699bf7794db3ca13/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/4694aa85f72f65761183c08893d6a7e0/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/948dcf4fee23a9f3b31eb40ce1877ee7/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/9e5da75179ec99e4199b8b5252a76cce/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/7310a5887f3b85a6d1798bcefed9b60f/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/caa43687ccb68af3b04921fbbe5416d7/libActonProject.a -ODebug -target x86_64-linux-gnu.2.27 -mcpu westmere -I /home/runner/work/sorespo/sorespo -I /home/runner/.cache/acton/zig-local-cache/o/e5b3852248c09411b6ccc1515bceb7e0 -I /home/runner/.cache/acton/zig-local-cache/o/b2b4c7ca963c150416155206af3f8490 -I /home/runner/.cache/acton/zig-local-cache/o/6b464c6629b432cab829f8899f398a4d -I /home/runner/.cache/acton/zig-local-cache/o/60d0b60d9ee3251ed5c75332b538e31b -I /home/runner/.cache/acton/zig-local-cache/o/daea5cee7ff7398524ad2fcf4b0068f9 -I /home/runner/.cache/acton/zig-local-cache/o/0ab51805a6f6801d558bbe7d65ee7a28 -I /home/runner/.cache/acton/zig-local-cache/o/b7ac9a4035471ecb755faa5cd88faef0 -Mroot /home/runner/.cache/acton/zig-local-cache/o/4694aa85f72f65761183c08893d6a7e0/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/948dcf4fee23a9f3b31eb40ce1877ee7/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/9e5da75179ec99e4199b8b5252a76cce/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/7310a5887f3b85a6d1798bcefed9b60f/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/caa43687ccb68af3b04921fbbe5416d7/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/33bd2e85dff30ed596634f2960bb097b/libbsdnt.a /home/runner/.cache/acton/zig-local-cache/o/7b1c3940fe0376624368e7460b70a0ec/libgc.a /home/runner/.cache/acton/zig-local-cache/o/c2ac5cb28989cd49f55df8a9f384d353/libmbedcrypto.a /home/runner/.cache/acton/zig-local-cache/o/0e7fb4f41c2d6d8e69614fa9fc10bedc/libmbedtls.a /home/runner/.cache/acton/zig-local-cache/o/5cd9f676c24a7a65952a64b3fe57a6da/libmbedx509.a /home/runner/.cache/acton/zig-local-cache/o/1cbda3a0c799e2e418482de549e8e076/libnetstring.a /home/runner/.cache/acton/zig-local-cache/o/0dee147c721d5f6ebc0bc4893a4f5a88/libpcre2-8.a /home/runner/.cache/acton/zig-local-cache/o/cd99b9977d422d8b7bbbf8667b4ecc73/libprotobuf-c.a /home/runner/.cache/acton/zig-local-cache/o/24e70a1d5a8f51eee2ea5a18c6199a7f/libsnappy-c.a /home/runner/.cache/acton/zig-local-cache/o/382511fb429af5e9898ae704f3c811ff/libtlsuv.a /home/runner/.cache/acton/zig-local-cache/o/c798586ac74c663d5086d5145e52aa23/libutf8proc.a /home/runner/.cache/acton/zig-local-cache/o/70a8a09fdeec881777490b98c5e3bc8c/libuv.a /home/runner/.cache/acton/zig-local-cache/o/3d329a51ccc308fcf5c68eed9d652001/libxml2.a /home/runner/.cache/acton/zig-local-cache/o/8b20e5edd995e19fedf0b642db33881b/libyyjson.a /home/runner/.cache/acton/zig-local-cache/o/5fd2355e2c4dd1884ca7e56a448d1af9/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/1f7fdbcda6073ab39cc70f73d9e4fbcd/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/948dcf4fee23a9f3b31eb40ce1877ee7/libActonProject.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/c2ac5cb28989cd49f55df8a9f384d353/libmbedcrypto.a /home/runner/.cache/acton/zig-local-cache/o/0e7fb4f41c2d6d8e69614fa9fc10bedc/libmbedtls.a /home/runner/.cache/acton/zig-local-cache/o/5cd9f676c24a7a65952a64b3fe57a6da/libmbedx509.a /home/runner/.cache/acton/zig-local-cache/o/70a8a09fdeec881777490b98c5e3bc8c/libuv.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a /home/runner/.cache/acton/zig-local-cache/o/b5a53c096ecf62db98c00b5113787dd8/libActon.a -lc++ -lc --cache-dir /home/runner/.cache/acton/zig-local-cache --global-cache-dir /home/runner/.cache/acton/zig-global-cache --name .test_sorespo.tmf --listen=- 
Build Summary: 66/69 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
+- install .test_sorespo.tmf transitive failure
   +- zig build-exe .test_sorespo.tmf Debug x86_64-linux-gnu.2.27 7 errors
error: the following build command failed with exit code 1:
/home/runner/.cache/acton/zig-local-cache/o/8f9bf560cf72ca2afb26d07314fa1d67/build /usr/lib/acton/zig/zig /home/runner/work/sorespo/sorespo /home/runner/.cache/acton/zig-local-cache /home/runner/.cache/acton/zig-global-cache --seed 0xac9c0544 -Z323aad234e120cc4 --prefix /home/runner/work/sorespo/sorespo/out --prefix-exe-dir bin -Dtarget=x86_64-linux-gnu.2.27 -Dcpu=westmere -Doptimize=Debug
```

The fix/workaround turned out to add `import testing` into `src/sorespo/tmf.act`. I'm sure a much smaller repro case exists, but it would be good to also/first understand why I could not even reproduce this locally.

### Expected Behavior

There should be no need to import modules that we're not calling?

--- a/compiler/actonc/Main.hs
+++ b/compiler/actonc/Main.hs
@@ -59,7 +59,6 @@ import Data.String.Utils (replace)
 import Data.Version (showVersion)
 import qualified Data.List
 import qualified Data.Map as M
-import qualified Data.Set
 import Error.Diagnose
 import Error.Diagnose.Style (defaultStyle)
 import qualified Filesystem.Path.CurrentOS as Fsco
@@ -490,38 +489,31 @@ findTyFile spaths mn = go spaths
         then return (Just fullPath)
         else go ps
 
-removeOrphanFiles :: FilePath -> IO ()
-removeOrphanFiles dir = do
-    -- Recursively get all files in the "out" directory.
+-- Remove orphaned files in out/types.
+-- Non-root files are removed if their module isn’t part of this build (i.e.
+-- no corresponding source .act in this build). The 'roots' argument lists the
+-- exact set of root stub files (*.root.c, *.test_root.c) to keep; others are
+-- removed.
+removeOrphanFiles :: Paths -> [CompileTask] -> [FilePath] -> IO ()
+removeOrphanFiles paths tasks roots = do
+    let dir = projTypes paths
     absOutFiles <- getFilesRecursive dir
-    let outFiles = map (makeRelative dir) absOutFiles
-
-    -- Map over each file.
-    forM_ outFiles $ \file -> do
-        -- Remove the file ending.
-        let fileNoExt = dropExtension (dropExtension file)
-            srcFile = ("src" </> fileNoExt <.> "act")
-
-        -- If the file is a .root.c file, always remove it and generate a new
-        -- one later, if necessary. Only an .act file with a root actor should
-        -- have a .root.c file but we cannot judge from here if the .act file
-        -- actually has a root actor, so the only safe choice is to remove the
-        -- file and let it be recreated if necessary. Should be very cheap
-        -- anyway since the file is so small. So check if filename ends with
-        -- ".root.c" and remove it!
-        if takeExtension file == ".c" && takeExtension (takeBaseName file) == ".root"
-          then do
-            removeIfExists (dir </> file)
-          else do
-            -- Check if there is a corresponding .act file in the "src" directory.
-            srcExists <- doesFileExist srcFile
-            -- If the .act file doesn't exist, remove the file in the "out" directory.
-            when (not srcExists) $ removeFile (dir </> file)
-              `catch` handleNotExists
+    let allowedBases = [ outBase paths (name t) | t <- tasks ]
+    forM_ absOutFiles $ \absFile -> do
+        let isC  = takeExtension absFile == ".c"
+            isH  = takeExtension absFile == ".h"
+            isTy = takeExtension absFile == ".ty"
+            bext = takeExtension (takeBaseName absFile)
+            isRootStub = isC && (bext == ".root" || bext == ".test_root")
+        if isRootStub
+          then when (not (absFile `elem` roots)) (removeIfExists absFile)
+          else when (isC || isH || isTy) $ do
+                 let base = dropExtension absFile
+                 unless (base `elem` allowedBases) (removeIfExists absFile)
   where
-        removeIfExists f = removeFile f `catch` handleNotExists
-        handleNotExists :: IOException -> IO ()
-        handleNotExists _ = return ()
+    removeIfExists f = removeFile f `catch` handleNotExists
+    handleNotExists :: IOException -> IO ()
+    handleNotExists _ = return ()
 
 
 
@@ -546,10 +538,7 @@ Terminology
   full TyTask where all module content is available.
 
 High-level Steps
-1) Orphan cleanup
-   - Remove stale files in out/ that don’t have corresponding source files.
-
-2) Discover and read tasks using header-first strategy (readModuleTask)
+1) Discover and read tasks using header-first strategy (readModuleTask)
    - For each module, try to use its .ty header to avoid parsing:
      - If .ty is missing/unreadable → parse .act to obtain imports (ActonTask).
      - If .ty exists and .act mtime <= .ty mtime → trust .ty header imports and
@@ -557,18 +546,18 @@ High-level Steps
      - If .act appears newer than .ty → verify by content hash:
        – If stored srcHash == current srcHash → header is still valid (TyTask)
        – Else → parse .act now to get accurate imports (ActonTask)
-   - This amounts to ensuring that the .ty is up to date with the .act source
-     file and reading from the .ty file header to get module imports etc, which
-     is much faster than parsing the .act file.
+   - This ensures that .ty is up to date with the .act source and lets us
+     read module imports/roots/docstring from the header, which is much faster
+     than parsing the .act file.
 
-3) Expand to include transitive project imports (readImports)
+2) Expand to include transitive project imports (readImports)
    - For project builds, we enumerate all modules under src/ upfront, so chasing
      is typically a no-op.
    - For single-file builds (or partial sets), we chase imports from the seeded
      tasks to ensure all project-local dependencies are included. We prefer .ty
      headers to avoid parsing; we parse only when a .ty is missing/unreadable.
 
-4) Decide freshness bottom-up (planModuleTasks)
+3) Decide freshness bottom-up (planModuleTasks)
    - Build the dependency graph using importsOf from tasks (TyTask uses header
      imports; ActonTask uses parsed AST imports). Compute a topological order.
    - For each module in order, fresh means: all project imports are already
@@ -577,12 +566,13 @@ High-level Steps
      sections. Stale modules become ActonTask (if not already) by parsing now.
    - With --verbose, we report which imports caused the staleness for clarity.
 
-5) Compile and build
-   - planModuleTasks returns modules in dependency (topological) order. compileTasks
-     trusts this order and compiles only ActonTask items (TyTask are cached).
-   - TyTask items remain lazy; code that needs only small bits (e.g., writeRootC)
-     reads roots from the header instead of forcing heavy loads.
-   - Final Zig link step executes unless --skip-build.
+4) Compile and build
+   - compileTasks compiles only ActonTask items (TyTask are cached).
+   - writeRootC writes per-binary root stubs (.root.c / .test_root.c) based on
+     the module’s .ty header roots.
+   - Right before invoking Zig, we prune root stubs using removeOrphanFiles so
+     only the discovered roots remain. After this point, removed stubs are NOT
+     regenerated in this build.
 
 ================================================================================
 -}
@@ -604,9 +594,6 @@ compileFiles gopts opts srcFiles = do
         putStrLn ("    binDir   : " ++ binDir paths)
         putStrLn ("    srcDir   : " ++ srcDir paths)
         iff (length srcFiles == 1) (putStrLn ("    modName  : " ++ prstr (modName paths)))
-
-    -- remove files in out that do not have corresponding source files!
-    removeOrphanFiles (projTypes paths)
 
     -- Build initial tasks using .ty headers when possible; parse only if .ty missing
     tasks0 <- mapM (readModuleTask gopts opts paths) srcFiles
@@ -1328,6 +1315,12 @@ zigBuild :: Acton.Env.Env0 -> C.GlobalOptions -> C.CompileOptions -> Paths -> [C
 zigBuild env gopts opts paths tasks binTasks = do
     allBinTasks <- mapM (writeRootC env gopts opts paths tasks) binTasks
     let realBinTasks = catMaybes allBinTasks
+    -- Clean out/types: drop stray root stubs and outputs that don’t belong
+    -- to the modules built in this run
+    let roots = [ if isTest bt then outBase paths (A.mname (rootActor bt)) ++ ".test_root.c"
+                                else outBase paths (A.mname (rootActor bt)) ++ ".root.c"
+                | bt <- realBinTasks ]
+    removeOrphanFiles paths tasks roots
     iff (not (quiet gopts opts)) $ putStrLn("  Final compilation step")
     timeStart <- getTime Monotonic
 

